src/main/java/core/utils/DateConverter.java

```diff
@@ -0,0 +1,118 @@
+package core.utils;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.StringTokenizer;
+import org.apache.commons.lang3.ArrayUtils;
+
+/**
+ * Transform a time phrase to a Date entity.
+ */
+public class DateConverter {
+
+    /**
+     * Transform a time phrase to a date entity.
+     *
+     * @param phrase is a string that we want to transform.
+     * @return a date entity.
+     */
+    public static Date transform(String phrase) {
+        if (isCorrect(phrase)) {
+            try {
+                return createDate(phrase);
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+        } else {
+            Time time = converter(phrase);
+            int valueUnity = TimeUnity.valueOf(time.getUnity()).getValue();
+            return calculate(valueUnity, time.getAmount());
+        }
+        return null;
+    }
+
+    /**
+     * Calculates a date given time units amount and type.
+     *
+     * @param election is an integer that represents type.
+     * @param value is an integer that represents amount.
+     * @return a date entity.
+     */
+    public static Date calculate(int election, int value) {
+        Date date = new Date();
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        calendar.add(election, value);
+        return date = calendar.getTime();
```

Comment:
You could directly return the value.

Reply:
Yeah it is redundant. I alreasy changed it.